src/legacy/core_plugins/kibana/public/discover/doc_table/lib/get_sort.js

```diff
@@ -29,24 +29,25 @@ function createSortObject(sortPair, indexPattern) {
     const [ field, direction ] = sortPair;
     return { [field]: direction };
   }
+  else if (_.isPlainObject(sortPair) && isSortable(Object.keys(sortPair)[0], indexPattern)) {
```

Comment:
I noticed a bug where `getSort.array` was [being used with an Elasticsearch formatted sort object](https://github.com/elastic/kibana/blob/e24e8291784189c8528a2e36fb1750befac5d2f0/src/legacy/core_plugins/kibana/public/discover/controllers/discover.js#L558) but we weren't actually handling that in the `getSort` code. So I updated this method to handle that case and added tests to make it more clear what all `getSort.array` is expected to do.